frescobaldi_app/vcs/gitjob.py

```diff
@@ -182,27 +182,24 @@ def _finished(self, exitcode, exitstatus):
     def stdout(self):
         """
         Returns the content of the stdout output, if any.
+        
+        Returns: 
+            None: Git() is running / Git() hasn't started / Git() has crashed
+             b'': Git() has finished running and returns a binary result.
+              []: Git() has finished running and returns a string-list result.
         """
-        # TODO: should we check isRunning() before or can we rely on the "is not None" check?
```

Comment:
`isRunning()` works by checking the process's state: `QProcess.NotRunning` 

There is a time interval between changing state into `QProcess.NotRunning`  and emitting `finished()` signal.  

`_stdout`  and  `_stderr` are updated after the signal `finished()` has been emitted.  So there will be the case in which `isRunning()` returns `False` while `_stdout` and `_stderr` are still `None`.